adhoc-controllers/main.go

```diff
@@ -65,8 +66,7 @@ func main() {
 
 	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
 		Scheme:                 scheme,
-		MetricsBindAddress:     metricsAddr,
-		Port:                   9443,
```

Comment:
Port was deprecated and removed in controller-runtime v0.16.0. Due to this breaking change we need to use another option struct. Instead, I found the default port is anyway 9443 ref: https://github.com/kubernetes-sigs/controller-runtime/blob/v0.16.0/pkg/webhook/server.go#L76-L78.

Reply:
can we expose that port via a flag so that user can use some other if required

Reply:
created #452
Will try to add a commit in this PR itself. Or we can merge this PR as it is and take this up later to avoid making the changes stale.

Reply:
we can consider shipping part of #452 issue